.circleci/workflows.yml

```diff
@@ -0,0 +1,1151 @@
+                    --skip-existing \
+                    "/tmp/workspace/generated-sql/sql/"
+                  PATH="venv/bin:$PATH" script/bqetl metadata update \
+                    --sql-dir /tmp/workspace/generated-sql/sql/ \
+                    /tmp/workspace/generated-sql/sql/
+            - persist_to_workspace:
+                root: /tmp/workspace
+                paths:
+                  - generated-sql
+      - run: exit 0
+  deploy-changes-to-stage:
+    docker: *docker
+    steps:
+      - when:
+          condition:
+            or:
+              - << pipeline.parameters.validate-sql >>
+              - << pipeline.parameters.validate-routines >>
+          steps:
+            - checkout
+            - *skip_forked_pr
+            - *restore_venv_cache
+            - *build
+            - *attach_generated_sql
+            - *copy_generated_sql
+            - add_ssh_keys:
+                fingerprints:
+                  - "22:b9:3c:1b:82:ab:3f:e4:b5:79:70:d1:7b:b9:28:d2"
+            - run:
+                name: Pull in generated-sql branch from remote
+                command: |
+                  ssh-keyscan -t rsa github.com >> ~/.ssh/known_hosts
+                  git clone --single-branch --branch generated-sql \
+                    git@github.com:mozilla/bigquery-etl \
+                    generated-sql
+            - run:
+                name: Deploy changes to stage
+                command: |
+                  if [ "<< pipeline.parameters.skip-stage-deploys >>" = "false" ]; then
+                    export GOOGLE_APPLICATION_CREDENTIALS="/tmp/gcp.json"
+                    echo "$GCLOUD_SERVICE_KEY" > "$GOOGLE_APPLICATION_CREDENTIALS"
+
+                    PATHS="$(git diff --no-index --name-only --diff-filter=d generated-sql/sql sql)" || true
+                    echo $PATHS
+                    PATH="venv/bin:$PATH" script/bqetl stage deploy \
+                      --dataset-suffix=$CIRCLE_SHA1 \
+                      --remove-updated-artifacts \
+                      $PATHS
+                  fi
+                # workaround for job failing with `Too long with no output (exceeded 10m0s): context deadline exceeded` error
+                no_output_timeout: 30m
+            - run:
+                name: Copy generated SQL to temporary stage directory
+                command: |
+                  mkdir -p /tmp/workspace/staged-generated-sql
+                  cp -r sql/ /tmp/workspace/staged-generated-sql/sql
+                  cp -r tests/ /tmp/workspace/staged-generated-sql/tests
+            - persist_to_workspace:
+                root: /tmp/workspace
+                paths:
+                  - staged-generated-sql
+      - run: exit 0
+  push-generated-sql:
+    docker: *docker
+    steps:
+      - when:
+          condition:
+            or:
+              - << pipeline.parameters.validate-sql >>
+              - << pipeline.parameters.validate-routines >>
```

Comment:
Since `push-generated-sql` should only happen on deployments, it seems like this should just be checking the `deploy` parameter (assuming you want to stick with that approach).